lib/pushmi_pullyu/aip/downloader.rb

```diff
@@ -1,215 +1,184 @@
 
     download_permissions
-
-    # Return directory name
-    PushmiPullyu::AIP.aip_directory(@noid)
   end
 
   private
 
-  def download_and_log(path_spec, local_path: nil, fedora_fetcher: nil)
-    # Fetcher is either for current object, or for a permission object
-    fedora_fetcher ||= @fedora_fetcher
-
-    # Sometimes we don't know filename in advance, so we use local_path ...
-    output_file = full_local_path(local_path || path_spec.local)
-    url = fedora_fetcher.object_url(path_spec.remote)
+  def download_and_log(path_spec, fedora_fetcher)
+    output_file = path_spec.local
 
-    log_fetching(path_spec, url, local_path: output_file)
+    log_fetching(fedora_fetcher.object_url(path_spec.remote), output_file)
 
-    rdf = (output_file =~ /\.n3$/)
+    is_rdf = (output_file !~ /\.n3$/)
 
-    success = fedora_fetcher.download_object(download_path: output_file,
-                                             url_extra: path_spec.remote,
-                                             optional: path_spec.optional,
-                                             rdf: rdf)
-    log_saved(path_spec, success, local_path: output_file)
+    is_success = fedora_fetcher.download_object(output_file,
+                                                url_extra: path_spec.remote,
+                                                optional: path_spec.optional,
+                                                is_rdf: is_rdf)
+    log_saved(is_success, output_file)
   end
 
   def download_permissions
     PushmiPullyu.logger.info("#{@noid}: looking up permissions from Solr ...")
     results = PushmiPullyu::AIP::SolrFetcher.new(@noid).fetch_permission_object_ids
     if results.empty?
       PushmiPullyu.logger.info("#{@noid}: permissions not found")
-      return
-    end
-    results.each do |permission_id|
-      PushmiPullyu.logger.info("#{@noid}: permission object #{permission_id} found")
-      download_permission(permission_id)
+    else
+      results.each do |permission_id|
+        PushmiPullyu.logger.info("#{@noid}: permission object #{permission_id} found")
+        download_permission(permission_id)
+      end
     end
   end
 
   def download_permission(permission_id)
-    permission_fetcher =
-      PushmiPullyu::AIP::FedoraFetcher.new(permission_id)
-
-    filename = permission_filename(permission_id)
-
     path_spec = OpenStruct.new(
       remote: nil,
-      local: filename,
+      local: "#{aip_dirs.metadata}/permission_#{permission_id}.n3",
       optional: false
     )
-    download_and_log(path_spec, fedora_fetcher: permission_fetcher)
+    download_and_log(path_spec, PushmiPullyu::AIP::FedoraFetcher.new(permission_id))
   end
 
   ### Logging
 
-  def log_fetching(path_spec, url, local_path: nil)
-    local_path ||= path_spec.local
-    message = "#{@noid}: #{local_path} -- fetching from #{url} ..."
-    PushmiPullyu::Logging.log_aip_activity(@noid, message)
+  def log_fetching(url, output_file)
```

Comment:
Simplfy both of these logging methods. No need for `local_path` anymore as `path_spec.local` will always have what we want. Reduce a lot of logic as a result.